azure-spring-boot-samples/azure-storage-spring-boot-sample/src/main/java/sample/storage/StorageSampleApplication.java

```diff
@@ -6,42 +6,81 @@
+import com.microsoft.azure.storage.blob.ServiceURL;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import org.apache.commons.io.FileUtils;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.CommandLineRunner;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
-import java.net.URISyntaxException;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
 
+@SuppressFBWarnings({"RV_RETURN_VALUE_IGNORED"})
 @SpringBootApplication
 public class StorageSampleApplication implements CommandLineRunner {
+    private static final String SOURCE_FILE = "storageTestFile.txt";
+
+    @Autowired
+    private ServiceURL serviceURL;
+
     @Autowired
-    private CloudStorageAccount cloudStorageAccount;
+    private ContainerURL containerURL;
+
+    @Autowired
+    private StorageProperties properties;
 
     public static void main(String[] args) {
         SpringApplication.run(StorageSampleApplication.class);
     }
 
-    public void run(String... var1) throws URISyntaxException, StorageException {
-        createContainerIfNotExists("mycontainer");
-    }
+    public void run(String... var1) throws IOException {
+        final File sourceFile = new File(this.getClass().getClassLoader().getResource(SOURCE_FILE).getFile());
+        final File downloadFile = Files.createTempFile("azure-storage-test", null).toFile();
+
+        StorageService.createContainer(containerURL,  properties.getContainerName());
+        final BlockBlobURL blockBlobURL = containerURL.createBlockBlobURL(SOURCE_FILE);
 
-    // Note: Here is the minimum sample code that demonstrates how CloudStorageAccount is autowired and used.
-    // For more complete Azure Storage API usage, please go to https://github.com/Azure-Samples and search repositories
-    // with key words `storage` and 'java'.
-    private void createContainerIfNotExists(String containerName) throws URISyntaxException, StorageException {
-        // Create the blob client.
-        final CloudBlobClient blobClient = cloudStorageAccount.createCloudBlobClient();
+        System.out.println("Enter a command:");
+        System.out.println("(P)utBlob | (G)etBlob | (D)eleteBlobs | (E)xitSample");
+        final BufferedReader reader =
+                new BufferedReader(new InputStreamReader(System.in, StandardCharsets.UTF_8.name()));
 
-        // Get a reference to a container.
-        // The container name must be lower case
-        final CloudBlobContainer container = blobClient.getContainerReference(containerName);
+        boolean isExit = false;
+        while (!isExit) {
+            System.out.println("Enter a command:");
+            final String input = reader.readLine();
+            if (input == null) {
```

Comment:
seems unnecessary, null will fall thought the default case.

Reply:
It's added because of a findbug warning..

Reply:
oh, I see, findbug is a noisy plugin indeed.